README.md

```diff
@@ -1,2 +1,99 @@
-# dropwizard-kafka-client-metrics
-Register your Kafka clients' built-in metrics to Dropwizard
+# kafka-dropwizard-reporter
+
+This package provides a `DropwizardReporter` class that connects the
+built-in metrics maintained by Kafka's client libraries with
+Dropwizard Metrics 3.0+.
+The Kafka metrics are added as `Gauge` instances to a Dropwizard
+`MetricRegistry` instance.
+
+If you're already using Dropwizard Metrics in your application
+to serve metrics via HTTP, Graphite, StatsD, etc.,
+this reporter provides an easy bridge to pass Kafka consumer,
+producer, and streams metrics to those same outputs.
+
+## Compatibility
+
+`dropwizard-metrics` 3.0 and above.
+
+`kafka-clients` 0.8.1 and above, including 0.9 and 0.10.
+Also functions with the new `kafka-streams` library in 0.10.0.0.
+
+## Usage
+
+First, declare a dependency on this package:
+```xml
+      <dependency>
+          <groupId>com.simple</groupId>
+          <artifactId>kafka-dropwizard-reporter</artifactId>
+          <version>1.0.0</version>
+      </dependency>
+```
+
+This package expects that `dropwizard-metrics` and `kafka-clients` libraries
+are also defined in your project.
+
+Then, include the `DropwizardReporter` class in the properties you pass
+to producers, consumers, and `KafkaStreams` applications:
+```
+metric.reporters=com.simple.metrics.kafka.DropwizardReporter
+```
+
+That client will now automatically register all of its built-in
+metrics with a Dropwizard `MetricRegistry` when it's initialized.
+
+## Integration with Dropwizard Applications
+
+The reporter calls `SharedMetricRegistries.getOrCreate("default")`
+to discover a registry. To make sure `DropwizardReporter` instances report
+to the main metrics registry of a Dropwizard application, call
+the following in your `run` method before defining any Kafka clients
+([this will be done by default in Dropwizard 1.0+](https://github.com/dropwizard/dropwizard/pull/1513)):
+```java
+public class MyApplication extends Application<MyConfiguration> {
+// [...]
+    @Override
+    public void run(MyConfiguration configuration, environment: Environment) {
+        SharedMetricRegistries.add("default", environment.metrics());
+    // [...]
+    }
+}
+```
+
+Metrics will be published with a prefix of
+`<metrics-prefix>.org.apache.kafka.common.metrics.MetricRegistry`
```

Comment:
I'm tempted by that shortcut, too. I was going to say that we should stick slavishly to the conventions of using full classpath in examples like https://github.com/dropwizard/metrics/blob/3.1-maintenance/metrics-logback/src/main/java/com/codahale/metrics/logback/InstrumentedAppender.java#L54 but I'm now realizing that our situation here is a bit different. In the InstrumentedAppender case, it's publishing metrics about what the base Appender class is doing. In our case, we're not publishing metrics about what MetricReporter is _doing_, but rather about metrics it's reporting.

Dropwizard's JVM metrics may be a better analogue, where they take metrics published as beans and republish them as Dropwizard metrics. In that case, the prefix is much shorter and doesn't represent a classpath.

If we're going to go with something that's not a full classpath, then I'd be for going all the way and using `<metrics-prefix>.org.apache.kafka`. I'd argue that the `common.metrics` piece is redundant; if we're republishing metrics, why care about where in the package structure the classes handling metrics are located?